engine/access/rest/select_filter_test.go

```diff
@@ -0,0 +1,144 @@
+package rest
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/onflow/flow-go/engine/access/rest/generated"
+)
+
+func ExampleSelectFilter() {
+
+	block, err := generateBlock()
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	selectKeys := []string{
+		"header.id",
```

Comment:
Could you add a case for top-level select field such as `payload` and at least one that is non-existing like `foo`?

Reply:
will add - but note, currently the select logic only supports leaf elements not top-level elements.

Reply:
https://github.com/dapperlabs/flow-go/issues/6034 added this issue and assigned it to myself